### ISSUE TITLE
 Support EigenStone Thermostat S2 (KTWKQ03ES)

### ISSUE THREAD

**Issue by slydiman** [rytilahti/python-miio#706]
Could you add support for  EigenStone Thermostat S2 (KTWKQ03ES), please?
The device id is lumi.airrtc.tcpecn02
I can help to test/debug.

**Comment by rytilahti** [comment]
Internet seems to suggest that it's not a wifi but a zigbee device. If it doesn't have wifi, it's out of scope of this project (although there has been some progress lately to control the gateway and its connected devices).

**Comment by slydiman** [comment]
Yes, it is ZigBee device and the gateway v2 supports this device in the MI Home app.
But the gateway reports in HA logs:
The device with sid 158d000xxxxxxx isn't supported of the used gateway firmware. Please update the gateway firmware if possible! This is the only way the issue can be solved.

It reported the same for Aqara 2ch relay, but I know it is fixed using other protocol via UDP port 54321.

**Comment by rytilahti** [comment]
There are some recent developments to add better support to gateway(s), but whether this can be supported with it is unclear and requires at least reversing how the communications are done. I wouldn't hold breath for getting these supported any time soon.

**Comment by rezmus** [comment]
unless you don't need events you can just control it regular miio way (pull props, invoke actions) sent to gateway device.
